<?xml version="1.0"?>
<robot name="my_2r">

  <material name="blue">
    <color rgba="0 0 0.8 1"/>
  </material>
  <material name="black">
    <color rgba="0 0 0 1"/>
  </material>
  <material name="white">
    <color rgba="1 1 1 1"/>
  </material>


  <link name="base_link">
    <visual>
      <geometry>
        <cylinder length="0" radius="0"/>
      </geometry>
    </visual>
  </link>

  <joint name="joint1" type="continuous">
    <axis xyz="0 0 1"/>
    <origin xyz="0 0 0" rpy="0 0 0"/>
    <parent link="base_link"/>
    <child link="link1"/>

  </joint>

  <link name="link1">
    <visual>
      <origin xyz="0.5 0 0" rpy="0 1.5708 0"/>
      <geometry>
        <cylinder length="1" radius="0.05"/>
      </geometry>
      <material name="blue"/>
    </visual>
  </link>

  <joint name="joint2" type="continuous">
    <axis xyz="0 0 1"/>
    <origin xyz="1 0 0" rpy="0 0 0"/>
    <parent link="link1"/>
    <child link="link2"/>
  </joint>

  <link name="link2">
    <visual>
      <origin rpy="0 1.5708 0" xyz=".5 0 0"/>
      <geometry>
        <cylinder length="1" radius="0.05"/>
      </geometry>
      <material name="white"/>
    </visual>
  </link>

  <joint name="end_effector_joint" type="fixed">
    <origin xyz="1 0 0" rpy="0 0 0"/>
    <parent link="link2"/>
    <child link="end_effector"/>
  </joint>

  <link name="end_effector">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0" radius="0"/>
      </geometry>
    </visual>
  </link>

</robot>
